cmd/sync/sync.go

```diff
@@ -75,6 +72,34 @@ func (f *syncFlags) syncOptionsFromArgs(cmd *cobra.Command, args []string) (*syn
 	return &opts, nil
 }
 
+func (f *syncFlags) syncOptions(cmd *cobra.Command, args []string, requiredMutators bundle.Mutator) (*sync.SyncOptions, error) {
+	// Try to get options from args first. If that fails, try to get them from bundle.
+	// If both fail, return the error from args (assuming users are trying to use the args
+	// version unless they explicitly use the bundle sync).
+	opts, argsErr := f.syncOptionsFromArgs(cmd, args)
+	if argsErr == nil {
+		return opts, nil
+	}
+	err := root.MustConfigureBundle(cmd, args)
+	if err != nil {
+		return nil, argsErr
+	}
+	b := bundle.GetOrNil(cmd.Context())
+	if b != nil {
+		// Apply the required mutators to make sure paths are set.
+		err = bundle.Apply(cmd.Context(), b, requiredMutators)
+		if err != nil {
+			return nil, argsErr
+		}
+		opts, err = f.syncOptionsFromBundle(cmd, b)
+		if err != nil {
+			return nil, argsErr
+		}
+		return opts, nil
+	}
```

Comment:
All of these error cases should trigger debug logs, because otherwise you run this from a bundle directory and are shown the help output thinking you're holding it wrong, while in fact something else is happening.

Therefore, I would rather not fallback a bunch of times, but instead have 2 clearly separate cases:
1. Either you're passing args and we're using them (exactly 2)
2. Or, you're running it from a bundle context and we expect everything to work; thus return bundle related errors.

You can implement (2) by using `TryConfigureBundle`, which doesn't return an error if the command is not executed in a bundle context, and then switch to the bundle configuration path where you return any errors you see directly to the user. You could also run `TryConfigureBundle` as a `PreRunE` function so that it is executed ALWAYS, and then use the presence or absence of a bundle on the context to figure out which of the two code paths to execute.